### DISCUSSION TITLE
Need support in regards to WhereUniqueInput

### DISCUSSION THREAD

**Discussion by martynball** [prisma/prisma#6753] [Q&A] [answered]
Hey, i'm trying to create a row within my `utm_groups` table if the `campaign`, `medium` and `source` is new, if not then return the existing group. However i'm getting an error on just the `utm_campaign` where at the moment.

> TS2322: Type '{ utm_campaigns: { name: any; }; utm_mediums: { name: any; }; utm_sources: { name: any; }; }' is not assignable to type 'utm_groupsWhereUniqueInput'.   Object literal may only specify known properties, and 'utm_campaigns' does not exist in type 'utm_groupsWhereUniqueInput'.

This is my code, should be rather simple, not too sure what the issue is:
```
const result = await prisma.journey_tracking.create({
  data: {
    journey_id: journey_identifier.journeys.id,
    utm_groups: {
      connectOrCreate: {
        where: {
          utm_campaigns: {name: data.utm_data.campaign},
          utm_mediums: {name: data.utm_data.medium},
          utm_sources: {name: data.utm_data.source}
        },
        create: {
          utm_campaigns: {
            connectOrCreate: {
              create: {name: data.utm_data.campaign},
              where: {name: data.utm_data.campaign}
            }
          },
          utm_mediums: {
            connectOrCreate: {
              create: {name: data.utm_data.medium},
              where: {name: data.utm_data.medium}
            }
          },
          utm_sources: {
            connectOrCreate: {
              create: {name: data.utm_data.source},
              where: {name: data.utm_data.source}
            }
          }
        }
      }
    },
  }
})
```

Here are the models:
```
model utm_groups {
  id               Int                @id @default(autoincrement()) @db.UnsignedInt
  utm_source_id    Int?               @db.UnsignedInt
  utm_medium_id    Int?               @db.UnsignedInt
  utm_campaign_id  Int?               @db.UnsignedInt
  utm_campaigns    utm_campaigns?     @relation(fields: [utm_campaign_id], references: [id])
  utm_mediums      utm_mediums?       @relation(fields: [utm_medium_id], references: [id])
  utm_sources      utm_sources?       @relation(fields: [utm_source_id], references: [id])
  journey_tracking journey_tracking[]

  @@unique([utm_source_id, utm_medium_id, utm_campaign_id], name: "utm_source_id_utm_medium_id_utm_campaign_id")
  @@index([utm_campaign_id], name: "FK_utm_groups_utm_campaigns")
  @@index([utm_medium_id], name: "FK_utm_groups_utm_mediums")
  @@index([utm_source_id], name: "FK_utm_groups_utm_sources")
}

model utm_campaigns {
  id         Int          @id @default(autoincrement()) @db.UnsignedInt
  name       String       @unique @db.VarChar(255)
  created_at DateTime     @default(now()) @db.Timestamp(0)
  utm_groups utm_groups[]
}

model utm_mediums {
  id         Int          @id @default(autoincrement()) @db.UnsignedInt
  name       String       @unique @db.VarChar(255)
  created_at DateTime     @default(now()) @db.Timestamp(0)
  utm_groups utm_groups[]
}

model utm_sources {
  id         Int          @id @default(autoincrement()) @db.UnsignedInt
  name       String       @unique @db.VarChar(255)
  created_at DateTime     @default(now()) @db.Timestamp(0)
  colour     String?      @db.VarChar(50)
  utm_groups utm_groups[]
}
```

**Comment by jherdman** [comment, marked as answer]
I think I've resolved this problem on my end. Bear in mind I'm very new to Prisma.

I did something sneaky to inspect the offending type (it's impenetrable, as you probably found too).

Here's my model:

```prisma
model Link {
  id         Int          @id @default(autoincrement())
  url        String       @unique
}
```

```typescript
import { Prisma } from '@prisma/client';

const wtf = Prisma.LinkWhereUniqueInput // this is my offending type -- same problem!
```

This showed me the following type:

```typescript
type Prisma.LinkWhereUniqueInput = {
  id?: number | undefined;
}
```

Note the missing `link` attribute.

It occurred to me that I added the `@unique` modifier after my initial migration, and therefore the generated types mustn't be correct.

The solution is to run `prisma generate`.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
